pkg/multicluster/virtual_cluster.go

```diff
@@ -62,12 +63,13 @@ func InitClusterInfo(cfg *rest.Config) error {
 		if err != nil {
 			return errors.Wrap(err, "fail to get registered clusters")
 		}
-		for _, cluster := range clusters.Items {
+
+		velaslices.ParMap(clusters.Items, func(cluster v1alpha1.VirtualCluster) error {
 			if err = SetClusterVersionInfo(ctx, cfg, cluster.Name); err != nil {
 				klog.Warningf("set cluster version for %s: %v, skip it...", cluster.Name, err)
-				continue
 			}
-		}
+			return nil
```

Comment:
If no error will be returned, maybe you can use ParFor instead of ParMap?
Another alternative solution is to return the error and run klog outside the ParMap, use AggregateError to merge the error array.